### ISSUE TITLE
Works great, but causes a "critical" error to be output

### ISSUE THREAD

**Issue by StopIteration** [hoyon/mpv-mpris#11]
This seems to be more of a mild annoyance than anything, but when mpv starts up I now get this error message:

    GLib-GIO-CRITICAL **: 21:03:15.407: g_dbus_connection_emit_signal: assertion 'G_IS_DBUS_CONNECTION (connection)' failed

It still works fine, and I'm kinda used to random glib spam now and then, but if possible it would still be nice to get rid of, if only to de-clutter the mpv cli.

**Comment by StopIteration** [comment]
I'm using 0.28.2 as well.

It happens only at startup, and in re-reproducing it myself, I've realized that it is also a very intermittent issue.  It seemed yesterday that it happened every time, but now that I'm trying to reproduce it, I've only gotten it to happen like twice (out of maybe 20+ attempts).  I'll let you know if I figure out how to consistently reproduce it, although it might just be that I notice it when it happens and it is actually quite rare.

**Comment by tsjnachos117** [comment]
I had a few glib messages at one point, when I realized I had the mpris.so file in ~/.config/mpv/scripts **AND** ~/.mpv/scripts. Maybe that's your issue too?

**Comment by 7ritn** [comment]
Okay, I think the problem was:
`// Register dbus object and interfaces
static void on_bus_acquired(GDBusConnection *connection,
                            G_GNUC_UNUSED const char *name,
                            gpointer user_data)
{
    GError *error = NULL;
    UserData *ud = user_data;
    ud->connection = connection;

    g_dbus_connection_register_object(connection, "/org/mpris/MediaPlayer2",
                                      ud->root_interface_info,
                                      &vtable_root,
                                      user_data, NULL, &error);
    if (error != NULL) {
        g_printerr("%s", error->message);
    }

    g_dbus_connection_register_object(connection, "/org/mpris/MediaPlayer2",
                                      ud->player_interface_info,
                                      &vtable_player,
                                      user_data, NULL, &error);
    if (error != NULL) {
        g_printerr("%s", error->message);
    }
}`

Here there is a complete code section a dublicate. I just removed the dublicate and the critical error disappeared.

**Comment by rien333** [comment]
I've been getting these warnings for at least a week or something (they went away for a bit, I think, after I started building this repo from its newest commit, but they've since returned. Its kind of annoying for cluttering up my terminal windows, but outside of that there is no noticeable effect I think.  Here is the output I'm getting (its not entirely consistent, but it's there in some form 95% of the time): 

 ``` console
$ mpv  --no-audio --no-video --no-resume-playback vid.mkv

(process:425681): GLib-CRITICAL **: 22:14:20.472: g_variant_new_object_path: assertion 'g_variant_is_object_path(object_path)' failed

(process:425681): GLib-CRITICAL **: 22:14:20.474: g_variant_dict_insert_value: assertion 'value != NULL' failed
     Video --vid=1 (*) (h264 1920x1040 23.976fps)
...
No video or audio streams selected.

Exiting... (Errors when loading file)

(process:425681): GLib-CRITICAL **: 22:14:20.492: g_variant_new_object_path: assertion 'g_variant_is_object_path(object_path)' failed

(process:425681): GLib-CRITICAL **: 22:14:20.492: g_variant_dict_insert_value: assertion 'value != NULL' failed
 ```

## Version info ##

```console
# build from source today
$ mpv --version
mpv 0.32.0-352-g530a0863b8 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on Sat Apr 11 20:27:18 CEST 2020
FFmpeg library versions:
   libavutil       56.42.102
   libavcodec      58.77.101
   libavformat     58.42.100
   libswscale      5.6.101
   libavfilter     7.77.101
   libswresample   3.6.100
FFmpeg version: N-97186-gda44bbefaa
$ uname -mor
5.6.3-arch1-1 x86_64 GNU/Linux
```

**Comment by hoyon** [comment]
Hmm that `g_variant_new_object_path` error is very strange. I've pushed up a new branch https://github.com/hoyon/mpv-mpris/tree/issue-11 which adds a debug log. Can you show what it outputs when you get that error?

**Comment by rien333** [comment]
This is a typical output:
```console
$ mpv --no-audio --no-video vid.mkv
An object is already exported for the interface org.mpris.MediaPlayer2 at /org/mpris/MediaPlayer2
(process:8173): GLib-GIO-CRITICAL **: 18:37:52.666: g_dbus_connection_register_object: assertion 'error == NULL || *error == NULL' failed
An object is already exported for the interface org.mpris.MediaPlayer2 at /org/mpris/MediaPlayer2
(process:8173): GLib-CRITICAL **: 18:37:52.675: g_variant_new_object_path: assertion 'g_variant_is_object_path (object_path)' failed

(process:8173): GLib-CRITICAL **: 18:37:52.675: g_variant_dict_insert_value: assertion 'value != NULL' failed
Resuming playback. This behavior can be disabled with --no-resume-playback.
/0
/0
     Video --vid=1 (*) (h264 1920x1040 23.976fps)
...
No video or audio streams selected.

Exiting... (Errors when loading file)

(process:8173): GLib-CRITICAL **: 18:37:52.688: g_variant_new_object_path: assertion 'g_variant_is_object_path (object_path)' failed

(process:8173): GLib-CRITICAL **: 18:37:52.688: g_variant_dict_insert_value: assertion 'value != NULL' failed
/-1

(process:8173): GLib-CRITICAL **: 18:37:52.689: g_variant_new_object_path: assertion 'g_variant_is_object_path (object_path)' failed

(process:8173): GLib-CRITICAL **: 18:37:52.689: g_variant_dict_insert_value: assertion 'value != NULL' failed
$ 
```

**Comment by hoyon** [comment]
Thanks! So there are two errors you've got in that output:

`An object is already exported ...`: You probably have mpv-mpris installed twice, which is what is causing the second attempt to load to fail. Can you check your `~/.config/mpv/scripts` and `/etc/mpv/scripts` directories to ensure you only have one copy.

`g_variant_new_object_path ...`: This was due to a special case where mpv returned -1 for `playlist-pos` which I wasn't aware of. I've handled this now in https://github.com/hoyon/mpv-mpris/commit/81e3de9107030eba735c927a4e2df8c5e7dd22c0 which should fix that.

**Comment by rien333** [comment]
Perfect, works great! Thank you so much for the fix.

**Comment by hoyon** [comment]
Great! I'll close this for now then. If there are is any more extraneous error logging feel free to create a new issue.

**Comment by tsjnachos117** [comment]
[Hoyon said](https://github.com/hoyon/mpv-mpris/issues/11#issuecomment-612654092)

>An object is already exported ...: You probably have mpv-mpris installed twice, which is what is causing the second attempt to load to fail. Can you check your ~/.config/mpv/scripts and /etc/mpv/scripts directories to ensure you only have one copy.

FYI, such a script could also be located in ~/.mpv. In fact, this was the only place in one's HOME directory to put scripts and configs for a while, IIRC.

**Comment by ammgws** [comment]
Another FYI:

I use the mpv-mpris AUR package and didn't realise it had changed from having the file in the user config to loading it by default from /etc: https://aur.archlinux.org/cgit/aur.git/commit/?h=mpv-mpris&id=da5d1f8f6e725bbd7f0bca897310bc1d5e93f8a4

Removed the script from my user config file and now the errors are gone.

**Comment by acarasimon96** [comment]
It looks like I'm getting this annoying behavior at random. It started right after I replaced dbus-daemon with dbus-broker in mid-January per [this announcement](https://archlinux.org/news/making-dbus-broker-our-default-d-bus-daemon/). I'll need to find out if bringing back dbus-broker is a valid workaround for this.

My distro is Manjaro Linux and I installed mpv-mpris from the extra repository. mpv version is 0.37.0.
